diff -ur mirall-1.6.1.orig/src/3rdparty/qtsingleapplication/qtlocalpeer.h mirall-1.6.1/src/3rdparty/qtsingleapplication/qtlocalpeer.h
--- mirall-1.6.1.orig/src/3rdparty/qtsingleapplication/qtlocalpeer.h	2014-07-02 12:45:20.573637070 +0300
+++ mirall-1.6.1/src/3rdparty/qtsingleapplication/qtlocalpeer.h	2014-07-02 12:45:23.797644648 +0300
@@ -27,7 +27,7 @@
 **
 ****************************************************************************/
 
-#include <QtSolutions/QtLockedFile>
+#include <qtlockedfile.h>
 
 #include <QLocalServer>
 #include <QLocalSocket>
diff -ur mirall-1.6.1.orig/src/CMakeLists.txt mirall-1.6.1/src/CMakeLists.txt
--- mirall-1.6.1.orig/src/CMakeLists.txt	2014-07-02 12:45:20.573637070 +0300
+++ mirall-1.6.1/src/CMakeLists.txt	2014-07-02 12:45:23.791644634 +0300
@@ -33,6 +33,10 @@
 endif()
 
 set(3rdparty_SRC
+3rdparty/qtsingleapplication/qtsingleapplication.cpp
+3rdparty/qtsingleapplication/qtlocalpeer.cpp
+3rdparty/qtsingleapplication/qtsinglecoreapplication.cpp
+3rdparty/qtlockedfile/qtlockedfile.cpp
 3rdparty/fancylineedit/fancylineedit.cpp
 3rdparty/QProgressIndicator/QProgressIndicator.cpp
 )
@@ -45,6 +49,9 @@
 endif()
 
 set(3rdparty_HEADER
+3rdparty/qtsingleapplication/qtlocalpeer.h
+3rdparty/qtsingleapplication/qtsingleapplication.h
+3rdparty/qtsingleapplication/qtsinglecoreapplication.h
 3rdparty/fancylineedit/fancylineedit.h
 3rdparty/QProgressIndicator/QProgressIndicator.h
 )
@@ -56,7 +63,15 @@
         )
 endif()
 
+if(NOT WIN32)
+	list(APPEND 3rdparty_SRC 3rdparty/qtlockedfile/qtlockedfile_unix.cpp)
+else()
+	list(APPEND 3rdparty_SRC 3rdparty/qtlockedfile/qtlockedfile_win.cpp )
+endif()
+
 set(3rdparty_INC
+    ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtlockedfile
+    ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtsingleapplication
     ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/QProgressIndicator
     ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/fancylineedit
     ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qjson
@@ -430,8 +445,6 @@
   target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} )
   target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
   target_link_libraries( ${APPLICATION_EXECUTABLE} updater )
-  target_link_libraries( ${APPLICATION_EXECUTABLE} QtSolutions_LockedFile-2.4 )
-  target_link_libraries( ${APPLICATION_EXECUTABLE} QtSolutions_SingleApplication-2.6 )
   target_link_libraries( ${APPLICATION_EXECUTABLE} ${OS_SPECIFIC_LINK_LIBRARIES} )
 
   install(TARGETS ${APPLICATION_EXECUTABLE}
diff -ur mirall-1.6.1.orig/src/mirall/application.cpp mirall-1.6.1/src/mirall/application.cpp
--- mirall-1.6.1.orig/src/mirall/application.cpp	2014-07-02 12:45:20.572637067 +0300
+++ mirall-1.6.1/src/mirall/application.cpp	2014-07-02 12:45:23.792644636 +0300
@@ -79,7 +79,7 @@
 // ----------------------------------------------------------------------------------
 
 Application::Application(int &argc, char **argv) :
-    QtSingleApplication(Theme::instance()->appName() ,argc, argv),
+    SharedTools::QtSingleApplication(Theme::instance()->appName() ,argc, argv),
     _gui(0),
     _theme(Theme::instance()),
     _helpOnly(false),
@@ -478,7 +478,7 @@
         return true;
     }
 
-    return QtSingleApplication::winEventFilter(pMsg, result);
+    return SharedTools::QtSingleApplication::winEventFilter(pMsg, result);
 }
 #endif
 
diff -ur mirall-1.6.1.orig/src/mirall/application.h mirall-1.6.1/src/mirall/application.h
--- mirall-1.6.1.orig/src/mirall/application.h	2014-07-02 12:45:20.572637067 +0300
+++ mirall-1.6.1/src/mirall/application.h	2014-07-02 12:45:23.792644636 +0300
@@ -20,7 +20,7 @@
 #include <QQueue>
 #include <QTimer>
 
-#include <QtSolutions/QtSingleApplication>
+#include "qtsingleapplication.h"
 
 #include "mirall/syncresult.h"
 #include "mirall/logbrowser.h"
@@ -40,7 +40,7 @@
 class SslErrorDialog;
 class SocketApi;
 
-class Application : public QtSingleApplication
+class Application : public SharedTools::QtSingleApplication
 {
     Q_OBJECT
 public:
