diff --git a/src/3rdparty/qtsingleapplication/qtlocalpeer.h b/src/3rdparty/qtsingleapplication/qtlocalpeer.h
index b28caad..a821030 100644
--- a/src/3rdparty/qtsingleapplication/qtlocalpeer.h
+++ b/src/3rdparty/qtsingleapplication/qtlocalpeer.h
@@ -28,7 +28,7 @@
 **
 **************************************************************************/
 
-#include "qtlockedfile.h"
+#include <QtSolutions/QtLockedFile>
 
 #include <QLocalServer>
 #include <QLocalSocket>
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6213e2c..a1f29a0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -30,33 +30,18 @@ if(SPARKLE_FOUND)
 endif(SPARKLE_FOUND)
 
 set(3rdparty_SRC
-3rdparty/qtsingleapplication/qtsingleapplication.cpp
-3rdparty/qtsingleapplication/qtlocalpeer.cpp
-3rdparty/qtsingleapplication/qtsinglecoreapplication.cpp
-3rdparty/qtlockedfile/qtlockedfile.cpp
 3rdparty/fancylineedit/fancylineedit.cpp
 3rdparty/QProgressIndicator/QProgressIndicator.cpp
 )
 
 set(3rdparty_HEADER
-3rdparty/qtsingleapplication/qtlocalpeer.h
-3rdparty/qtsingleapplication/qtsingleapplication.h
-3rdparty/qtsingleapplication/qtsinglecoreapplication.h
 3rdparty/fancylineedit/fancylineedit.h
 3rdparty/QProgressIndicator/QProgressIndicator.h
 )
 
 qt_wrap_cpp(3rdparty_MOC ${3rdparty_HEADER})
 
-if(NOT WIN32)
-	list(APPEND 3rdparty_SRC 3rdparty/qtlockedfile/qtlockedfile_unix.cpp)
-else()
-	list(APPEND 3rdparty_SRC 3rdparty/qtlockedfile/qtlockedfile_win.cpp )
-endif()
-
 set(3rdparty_INC
-    ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtlockedfile
-    ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtsingleapplication
     ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/QProgressIndicator
     ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/fancylineedit
     ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qjson
@@ -415,6 +400,8 @@ set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} updater )
+target_link_libraries( ${APPLICATION_EXECUTABLE} QtSolutions_LockedFile-2.4 )
+target_link_libraries( ${APPLICATION_EXECUTABLE} QtSolutions_SingleApplication-2.6 )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${OS_SPECIFIC_LINK_LIBRARIES} )
 
 install(TARGETS ${APPLICATION_EXECUTABLE}
diff --git a/src/mirall/application.cpp b/src/mirall/application.cpp
index 6c51aac..8347398 100644
--- a/src/mirall/application.cpp
+++ b/src/mirall/application.cpp
@@ -79,7 +79,7 @@ QString applicationTrPath()
 // ----------------------------------------------------------------------------------
 
 Application::Application(int &argc, char **argv) :
-    SharedTools::QtSingleApplication(argc, argv),
+    QtSingleApplication(argc, argv),
     _gui(0),
     _theme(Theme::instance()),
     _helpOnly(false),
@@ -472,7 +472,7 @@ bool Application::winEventFilter(MSG *pMsg, long *result)
         return true;
     }
 
-    return SharedTools::QtSingleApplication::winEventFilter(pMsg, result);
+    return QtSingleApplication::winEventFilter(pMsg, result);
 }
 #endif
 
diff --git a/src/mirall/application.h b/src/mirall/application.h
index 95939e8..c699946 100644
--- a/src/mirall/application.h
+++ b/src/mirall/application.h
@@ -19,7 +19,7 @@
 #include <QPointer>
 #include <QQueue>
 
-#include "qtsingleapplication.h"
+#include <QtSolutions/QtSingleApplication>
 
 #include "mirall/syncresult.h"
 #include "mirall/logbrowser.h"
@@ -37,7 +37,7 @@ class Folder;
 class SslErrorDialog;
 class SocketApi;
 
-class Application : public SharedTools::QtSingleApplication
+class Application : public QtSingleApplication
 {
     Q_OBJECT
 public:
