diff -ur owncloudclient-2.0.1.orig/src/cmd/CMakeLists.txt owncloudclient-2.0.1/src/cmd/CMakeLists.txt
--- owncloudclient-2.0.1.orig/src/cmd/CMakeLists.txt	2015-09-08 19:26:20.631229885 +0300
+++ owncloudclient-2.0.1/src/cmd/CMakeLists.txt	2015-09-08 19:36:50.369513888 +0300
@@ -23,7 +23,7 @@
 	set_target_properties(${cmd_NAME} PROPERTIES
 	        RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY} )
         set_target_properties(${cmd_NAME} PROPERTIES
-		INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
+		INSTALL_RPATH "${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
 
 	target_link_libraries(${cmd_NAME} ${synclib_NAME})
 endif()
diff -ur owncloudclient-2.0.1.orig/src/crashreporter/CMakeLists.txt owncloudclient-2.0.1/src/crashreporter/CMakeLists.txt
--- owncloudclient-2.0.1.orig/src/crashreporter/CMakeLists.txt	2015-09-08 19:26:20.631229885 +0300
+++ owncloudclient-2.0.1/src/crashreporter/CMakeLists.txt	2015-09-08 19:37:14.124735192 +0300
@@ -36,7 +36,7 @@
 
     set_target_properties(${CRASHREPORTER_EXECUTABLE} PROPERTIES AUTOMOC ON)
     set_target_properties(${CRASHREPORTER_EXECUTABLE} PROPERTIES RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY} )
-    set_target_properties(${CRASHREPORTER_EXECUTABLE} PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
+    set_target_properties(${CRASHREPORTER_EXECUTABLE} PROPERTIES INSTALL_RPATH "${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
     target_link_libraries(${CRASHREPORTER_EXECUTABLE}
         crashreporter-gui
         ${QT_LIBRARIES}
diff -ur owncloudclient-2.0.1.orig/src/gui/CMakeLists.txt owncloudclient-2.0.1/src/gui/CMakeLists.txt
--- owncloudclient-2.0.1.orig/src/gui/CMakeLists.txt	2015-09-08 19:26:20.632229887 +0300
+++ owncloudclient-2.0.1/src/gui/CMakeLists.txt	2015-09-08 19:38:05.229865174 +0300
@@ -261,7 +261,7 @@
 )
 # Only relevant for Linux? On OS X it by default properly checks in the bundle directory next to the exe
 set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
-	INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
+	INSTALL_RPATH "${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
 
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
diff -ur owncloudclient-2.0.1.orig/src/libsync/CMakeLists.txt owncloudclient-2.0.1/src/libsync/CMakeLists.txt
--- owncloudclient-2.0.1.orig/src/libsync/CMakeLists.txt	2015-09-08 19:26:20.635229893 +0300
+++ owncloudclient-2.0.1/src/libsync/CMakeLists.txt	2015-09-08 19:38:22.900903722 +0300
@@ -158,7 +158,7 @@
 	RUNTIME_OUTPUT_DIRECTORY ${BIN_OUTPUT_DIRECTORY}
 )
 set_target_properties( ${synclib_NAME} PROPERTIES
-	INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
+	INSTALL_RPATH "${LIB_INSTALL_DIR}/${APPLICATION_EXECUTABLE}" )
 
 target_link_libraries(${synclib_NAME} ${libsync_LINK_TARGETS} )
 
