diff -ur owncloudclient-2.0.1.orig/src/3rdparty/qtsingleapplication/qtlocalpeer.h owncloudclient-2.0.1/src/3rdparty/qtsingleapplication/qtlocalpeer.h
--- owncloudclient-2.0.1.orig/src/3rdparty/qtsingleapplication/qtlocalpeer.h	2015-09-08 19:26:20.625229873 +0300
+++ owncloudclient-2.0.1/src/3rdparty/qtsingleapplication/qtlocalpeer.h	2015-09-08 19:26:42.162924668 +0300
@@ -27,7 +27,7 @@
 **
 ****************************************************************************/
 
-#include <qtlockedfile.h>
+#include <QTSolutions/qtlockedfile.h>
 
 #include <QLocalServer>
 #include <QLocalSocket>
diff -ur owncloudclient-2.0.1.orig/src/gui/application.cpp owncloudclient-2.0.1/src/gui/application.cpp
--- owncloudclient-2.0.1.orig/src/gui/application.cpp	2015-09-08 19:26:20.632229887 +0300
+++ owncloudclient-2.0.1/src/gui/application.cpp	2015-09-08 19:30:46.836978916 +0300
@@ -84,7 +84,7 @@
 // ----------------------------------------------------------------------------------
 
 Application::Application(int &argc, char **argv) :
-    SharedTools::QtSingleApplication(Theme::instance()->appName() ,argc, argv),
+    QtSingleApplication(Theme::instance()->appName() ,argc, argv),
     _gui(0),
     _theme(Theme::instance()),
     _helpOnly(false),
diff -ur owncloudclient-2.0.1.orig/src/gui/application.h owncloudclient-2.0.1/src/gui/application.h
--- owncloudclient-2.0.1.orig/src/gui/application.h	2015-09-08 19:26:20.632229887 +0300
+++ owncloudclient-2.0.1/src/gui/application.h	2015-09-08 19:32:24.745101837 +0300
@@ -20,7 +20,7 @@
 #include <QQueue>
 #include <QTimer>
 
-#include "qtsingleapplication.h"
+#include "QtSolutions/QtSingleApplication"
 
 #include "syncresult.h"
 #include "logbrowser.h"
@@ -47,7 +47,7 @@
  * @brief The Application class
  * @ingroup gui
  */
-class Application : public SharedTools::QtSingleApplication
+class Application : public QtSingleApplication
 {
     Q_OBJECT
 public:
diff -ur owncloudclient-2.0.1.orig/src/gui/CMakeLists.txt owncloudclient-2.0.1/src/gui/CMakeLists.txt
--- owncloudclient-2.0.1.orig/src/gui/CMakeLists.txt	2015-09-08 19:26:20.632229887 +0300
+++ owncloudclient-2.0.1/src/gui/CMakeLists.txt	2015-09-08 19:29:09.912584234 +0300
@@ -120,10 +120,6 @@
 set(3rdparty_SRC
     ../3rdparty/fancylineedit/fancylineedit.cpp
     ../3rdparty/QProgressIndicator/QProgressIndicator.cpp
-    ../3rdparty/qtlockedfile/qtlockedfile.cpp
-    ../3rdparty/qtsingleapplication/qtlocalpeer.cpp
-    ../3rdparty/qtsingleapplication/qtsingleapplication.cpp
-    ../3rdparty/qtsingleapplication/qtsinglecoreapplication.cpp
     ../3rdparty/certificates/p12topem.cpp
    )
 
@@ -135,19 +131,11 @@
         )
 endif()
 
-if(NOT WIN32)
-   list(APPEND 3rdparty_SRC ../3rdparty/qtlockedfile/qtlockedfile_unix.cpp)
-else()
-   list(APPEND 3rdparty_SRC ../3rdparty/qtlockedfile/qtlockedfile_win.cpp )
-endif()
-
 set(3rdparty_INC
     ${CMAKE_SOURCE_DIR}/src/3rdparty/fancylineedit
     ${CMAKE_SOURCE_DIR}/src/3rdparty/qjson
     ${CMAKE_SOURCE_DIR}/src/3rdparty/QProgressIndicator
-    ${CMAKE_SOURCE_DIR}/src/3rdparty/qtlockedfile
     ${CMAKE_SOURCE_DIR}/src/3rdparty/qtmacgoodies/src
-    ${CMAKE_SOURCE_DIR}/src/3rdparty/qtsingleapplication
     )
 
 include_directories(${3rdparty_INC})
@@ -266,6 +254,8 @@
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} updater )
+target_link_libraries( ${APPLICATION_EXECUTABLE} QtSolutions_LockedFile-2.4 )
+target_link_libraries( ${APPLICATION_EXECUTABLE} QtSolutions_SingleApplication-2.6 )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${OS_SPECIFIC_LINK_LIBRARIES} )
 
 if(WITH_CRASHREPORTER)
