diff -ur owncloudclient-2.1.0.orig/src/3rdparty/qtsingleapplication/qtlocalpeer.h owncloudclient-2.1.0/src/3rdparty/qtsingleapplication/qtlocalpeer.h
--- owncloudclient-2.1.0.orig/src/3rdparty/qtsingleapplication/qtlocalpeer.h	2015-12-03 10:19:51.000000000 +0100
+++ owncloudclient-2.1.0/src/3rdparty/qtsingleapplication/qtlocalpeer.h	2015-12-04 11:44:30.631090122 +0100
@@ -27,7 +27,7 @@
 **
 ****************************************************************************/
 
-#include <qtlockedfile.h>
+#include <qt5/QTSolutions/qtlockedfile.h>
 
 #include <QLocalServer>
 #include <QLocalSocket>
diff -ur owncloudclient-2.1.0.orig/src/gui/application.cpp owncloudclient-2.1.0/src/gui/application.cpp
--- owncloudclient-2.1.0.orig/src/gui/application.cpp	2015-12-03 10:19:51.000000000 +0100
+++ owncloudclient-2.1.0/src/gui/application.cpp	2015-12-04 11:45:18.953025851 +0100
@@ -92,7 +92,7 @@
 // ----------------------------------------------------------------------------------
 
 Application::Application(int &argc, char **argv) :
-    SharedTools::QtSingleApplication(Theme::instance()->appName() ,argc, argv),
+    QtSingleApplication(Theme::instance()->appName() ,argc, argv),
     _gui(0),
     _theme(Theme::instance()),
     _helpOnly(false),
diff -ur owncloudclient-2.1.0.orig/src/gui/application.h owncloudclient-2.1.0/src/gui/application.h
--- owncloudclient-2.1.0.orig/src/gui/application.h	2015-12-03 10:19:51.000000000 +0100
+++ owncloudclient-2.1.0/src/gui/application.h	2015-12-04 11:46:13.820817409 +0100
@@ -22,7 +22,7 @@
 #include <QElapsedTimer>
 #include <QNetworkConfigurationManager>
 
-#include "qtsingleapplication.h"
+#include "qt5/QtSolutions/QtSingleApplication"
 
 #include "syncresult.h"
 #include "logbrowser.h"
@@ -49,7 +49,7 @@
  * @brief The Application class
  * @ingroup gui
  */
-class Application : public SharedTools::QtSingleApplication
+class Application : public QtSingleApplication
 {
     Q_OBJECT
 public:
diff -ur owncloudclient-2.1.0.orig/src/gui/CMakeLists.txt owncloudclient-2.1.0/src/gui/CMakeLists.txt
--- owncloudclient-2.1.0.orig/src/gui/CMakeLists.txt	2015-12-03 10:19:51.000000000 +0100
+++ owncloudclient-2.1.0/src/gui/CMakeLists.txt	2015-12-04 11:48:31.325788916 +0100
@@ -135,10 +135,6 @@
 set(3rdparty_SRC
     ../3rdparty/fancylineedit/fancylineedit.cpp
     ../3rdparty/QProgressIndicator/QProgressIndicator.cpp
-    ../3rdparty/qtlockedfile/qtlockedfile.cpp
-    ../3rdparty/qtsingleapplication/qtlocalpeer.cpp
-    ../3rdparty/qtsingleapplication/qtsingleapplication.cpp
-    ../3rdparty/qtsingleapplication/qtsinglecoreapplication.cpp
     ../3rdparty/certificates/p12topem.cpp
    )
 
@@ -150,19 +146,11 @@
         )
 endif()
 
-if(NOT WIN32)
-   list(APPEND 3rdparty_SRC ../3rdparty/qtlockedfile/qtlockedfile_unix.cpp)
-else()
-   list(APPEND 3rdparty_SRC ../3rdparty/qtlockedfile/qtlockedfile_win.cpp )
-endif()
-
 set(3rdparty_INC
     ${CMAKE_SOURCE_DIR}/src/3rdparty/fancylineedit
     ${CMAKE_SOURCE_DIR}/src/3rdparty/qjson
     ${CMAKE_SOURCE_DIR}/src/3rdparty/QProgressIndicator
-    ${CMAKE_SOURCE_DIR}/src/3rdparty/qtlockedfile
     ${CMAKE_SOURCE_DIR}/src/3rdparty/qtmacgoodies/src
-    ${CMAKE_SOURCE_DIR}/src/3rdparty/qtsingleapplication
     )
 
 include_directories(${3rdparty_INC})
@@ -281,6 +269,8 @@
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} updater )
+target_link_libraries( ${APPLICATION_EXECUTABLE} Qt5Solutions_LockedFile-2.4 )
+target_link_libraries( ${APPLICATION_EXECUTABLE} Qt5Solutions_SingleApplication-2.6 )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${OS_SPECIFIC_LINK_LIBRARIES} )
 
 if(WITH_CRASHREPORTER)
